src/Symfony/Component/Lock/Store/MongoDbStore.php

```diff
@@ -0,0 +1,435 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock\Store;
+
+use BadMethodCallException;
+use MongoDB\BSON\UTCDateTime;
+use MongoDB\Client;
+use MongoDB\Collection;
+use MongoDB\Driver\Command;
+use MongoDB\Driver\Exception\WriteException;
+use MongoDB\Driver\Manager;
+use MongoDB\Exception\DriverRuntimeException;
+use MongoDB\Exception\InvalidArgumentException as MongoInvalidArgumentException;
+use MongoDB\Exception\UnsupportedException;
+use Symfony\Component\Lock\Exception\InvalidArgumentException;
+use Symfony\Component\Lock\Exception\LockAcquiringException;
+use Symfony\Component\Lock\Exception\LockConflictedException;
+use Symfony\Component\Lock\Exception\LockExpiredException;
+use Symfony\Component\Lock\Exception\LockStorageException;
+use Symfony\Component\Lock\Exception\NotSupportedException;
+use Symfony\Component\Lock\Key;
+use Symfony\Component\Lock\StoreInterface;
+
+/**
+ * MongoDbStore is a StoreInterface implementation using MongoDB as a storage
+ * engine.
+ *
+ * @author Joe Bennett <joe@assimtech.com>
+ */
+class MongoDbStore implements StoreInterface
+{
+    private $mongo;
+    private $options;
+    private $initialTtl;
+
+    private $collection;
+    private $databaseVersion;
+    private $manager;
+
+    /**
+     * @param Client $mongo
+     * @param array  $options    See below
+     * @param float  $initialTtl The expiration delay of locks in seconds
+     *
+     * @throws InvalidArgumentException if required options are not provided
+     *
+     * Options:
+     *      database:                  The name of the database [required]
+     *      collection:                The name of the collection [default: lock]
+     *      createTtlIndexProbability: Should a TTL Index be created expressed as a probability from 0.0 to 1.0 [default: 0.001]
```

Comment:
This option is used both for creating index and removing manually expired locks (depends on the version). 
What's about `gcProbablity` like in PdoStore?

Moreover, I wonder if ti make send to create several time the TTL index...

Reply:
I changed to gcProbability.

Re creating the TTL Index several times, in Mongo most actions are usually an enforcement. There used to be a method called `ensureIndex` however it was deprecated in favour of just calling `createIndex` see: https://docs.mongodb.com/manual/reference/method/db.collection.ensureIndex/